ogre2/include/ignition/rendering/ogre2/Ogre2RenderEngine.hh

```diff
@@ -232,6 +232,9 @@ namespace ignition
       /// \brief True to use the current opengl context
       private: bool useCurrentGLContext = false;
 
+      /// \brief True to use the Metal render system
+      private: bool useMetalRenderSystem = false;
```

Comment:
I think adding a new variable here breaks ABI. Can this be moved to `Ogre2RenderEnginePrivate`?

Reply:
It would also make more sense if it's changed to a more generic enum.

In the future there will be Vulkan support too (and possibly D3D11)

Should the enum include d3d & vk even if not yet implemented?

Reply:
Yes, this would also be useful to the work enabling Metal in MinimalScene. Any suggestions where to add enums? (do not want to break ABI if this is going into ign-rendering6).

Reply:
Moved in 25d8719d. Will add enum in separate commit.

Reply:
>  this would also be useful to the work enabling Metal in MinimalScene

Just looking at the current implementation and the [MinimalScene in your ign-gui branch](https://github.com/srmainwaring/ign-gui/blob/feature/ign-gui7-metal/src/plugins/minimal_scene/MinimalScene.cc), it's specifying `metal` in the params as string right?  If you need to expose this enum somehow, you can add it to RenderEngine.hh since adding a new enum should not break ABI according to [this page](https://community.kde.org/Policies/Binary_Compatibility_Issues_With_C%2B%2B). Otherwise, we can just add the enum in this class given that currently the ogre2 engine in ign-rendering is currently the only one that has support for multiple backends.

Reply:
Thanks for the ABI link. The string `metal` is used in the params string for the render engine, but also in a number of conditional expressions. An enum would be better for the latter.

I've added an enum class `GraphicsAPI` and a utility `GraphicsAPIUtils` for conversion to and from `std::string`. I've taken a steer from Qt (`QSGRendererInterface`) in choosing names and used the `IGN_ENUM` macro for the implementation. I opted to place the enum in its own file rather than modify any existing class files.